arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -38,3 +38,10 @@ config NXP_MPU
 	default n
 	help
 	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
+	bool "Add MPU access to write to flash"
+	depends on NXP_MPU
+	default n
```

Comment:
Also, at least the k6x devices mentioned in the commit text have the ability to write protect flash regions, which is what will be needed to actually provide secure boot in a meaningful way.